File: app/src/main/java/poker/fivecardstats/activity/MainActivity.java
package poker.fivecardstats.activity;

import android.app.AlertDialog;
import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.AdapterView;
import android.widget.EditText;
import android.widget.ListView;

import java.util.List;

import poker.fivecardstats.R;
import poker.fivecardstats.adapter.UsersAdapter;
import poker.fivecardstats.dao.UserDataSource;
import poker.fivecardstats.model.User;

public class MainActivity extends AppCompatActivity {
    private FloatingActionButton fab;
    private UserDataSource mUserDataSource;
    private ListView listView;
    private List<User> mUsers;
    private  UsersAdapter mAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initDB();
        init();
        listViewManager();

        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        fabManager();
    }

    private void fabManager() {
        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                LayoutInflater layoutInflater = LayoutInflater.from(MainActivity.this);

                final View inflator = layoutInflater.inflate(R.layout.content_add_player, null);

                AlertDialog.Builder builder = new AlertDialog.Builder(MainActivity.this);
                builder.setTitle("Add new person");
                builder.setView(inflator);

                final EditText etName = (EditText) inflator.findViewById(R.id.et_name);

                builder.setPositiveButton("Submit", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        String name = etName.getText().toString();

                        mUserDataSource.create(name, 0);
                        updateList();
                    }
                }).show();
            }
        });
    }

    private void listViewManager() {
        mUsers = mUserDataSource.getUsers();
        mAdapter = new UsersAdapter(this, mUsers);
        listView.setAdapter(mAdapter);
        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                User u = mUsers.get(position);

                Intent score = new Intent(MainActivity.this, ScoreActivity.class);
                score.putExtra("id", u.getId());
                startActivity(score);
            }
        });
    }

    private void init() {
        fab = (FloatingActionButton) findViewById(R.id.fab);
        listView = (ListView) findViewById(R.id.users_listview);
    }

    private void initDB() {
        mUserDataSource = new UserDataSource(this);
        try {
            mUserDataSource.open();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    private void updateList() {
        mAdapter.clear();

        mUsers = mUserDataSource.getUsers();
        mAdapter = new UsersAdapter(this, mUsers);
        listView.setAdapter(mAdapter);

        mAdapter.notifyDataSetChanged();
    }

    @Override
    public void onResume() {
        super.onResume();
        updateList();
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return false;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return false;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/kotlin/poker/fivecardstats/activity/MainActivity.kt
package poker.fivecardstats.activity

import android.app.AlertDialog
import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.Menu
import android.view.MenuItem
import android.widget.Button
import android.widget.EditText
import android.widget.ListView
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import com.google.android.material.floatingactionbutton.FloatingActionButton
import poker.fivecardstats.R
import poker.fivecardstats.adapter.UsersAdapter
import poker.fivecardstats.dao.UserDataSource

class MainActivity : AppCompatActivity() {
    private lateinit var fab: FloatingActionButton
    private lateinit var mUserDataSource: UserDataSource
    private lateinit var listView: ListView
    private lateinit var mAdapter: UsersAdapter
    private var TOTAL_PLAYERS = 1

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)

        initDB()
        setupListView() // Setup UI with an empty list
        fabManager()
        clearButtonOnClick()
    }

    private fun clearButtonOnClick() {
        val clear = findViewById<Button>(R.id.clear_button)
        clear.setOnClickListener {
            mUserDataSource.resetAllScores()
            updateList() // Refresh the list to show updated scores
        }
    }

    private fun fabManager() {
        fab = findViewById(R.id.fab)
        fab.setOnClickListener {
            val layoutInflater = LayoutInflater.from(this@MainActivity)
            val inflator = layoutInflater.inflate(R.layout.content_add_player, null)

            val builder = AlertDialog.Builder(this@MainActivity)
            builder.setTitle("Add new person")
            builder.setView(inflator)

            val etName = inflator.findViewById<EditText>(R.id.et_name)

            builder.setPositiveButton("Submit") { _, _ ->
                var name = etName.text.toString()

                if (name.isEmpty()) {
                    name = "Player$TOTAL_PLAYERS"
                    TOTAL_PLAYERS++
                }

                mUserDataSource.create(name)
                updateList() // Refresh the list to show the new user
            }.show()
        }
    }

    private fun setupListView() {
        listView = findViewById(R.id.users_listview)
        mAdapter = UsersAdapter(this, arrayListOf()) // Start with an empty list
        listView.adapter = mAdapter
        listView.setOnItemClickListener { _, _, position, _ ->
            val user = mAdapter.getItem(position) ?: return@setOnItemClickListener

            val userIntent = Intent(this@MainActivity, UserActivity::class.java).apply {
                putExtra("id", user.id)
                putExtra("name", user.name)
            }
            startActivity(userIntent)
        }
    }

    private fun initDB() {
        mUserDataSource = UserDataSource(this)
        try {
            mUserDataSource.open()
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    private fun updateList() {
        val users = mUserDataSource.users
        mAdapter.clear()
        mAdapter.addAll(users)
    }

    public override fun onResume() {
        super.onResume()
        updateList() // Load data when the activity becomes visible
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_main, menu)
        return false
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return if (item.itemId == R.id.action_settings) {
            false
        } else {
            super.onOptionsItemSelected(item)
        }
    }
}